streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSource.java

```diff
@@ -68,6 +68,10 @@ public void materialize() {
         this.queryableName = storeName;
     }
 
+    public boolean materialized() {
```

Comment:
In `TableSourceNode` and `StreamToTableNode` we use `if (ktableSource. queryableName() != null)` to make a decision about materialization -- given this new method, can we update both to also use this new method?

Nit: should the method be called `needsMaterialization()` ? The flag does not indicate that the table _is_ materialized but that it _must_ be materialized?

Reply:
Good idea! I've added that in the latest commit. I decided not to change the method name because it gets into whether `materialize()` should be like `requestMaterialization()`, etc, but that's a bigger change. `void materialize()` and `boolean materialized()` seem like a good pair.